Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingCustomMesh.cs

```diff
@@ -43,18 +51,109 @@ public bool DrawProcessedMesh
             }
         }
 
+        /// <summary>
+        /// Indicates if the previous import is still being processed.
+        /// </summary>
+        public bool isImportActive { get; private set; }
+
+        /// <summary>
+        /// The material to use if rendering.
+        /// </summary>
         protected override Material RenderMaterial { get { return MeshMaterial; } }
         
         /// <summary>
-        /// World anchor used by the custom mesh
+        /// To prevent us from importing too often, we keep track of the last import.
         /// </summary>
-        public WorldAnchor LocalWorldAnchor { get; private set; }
-
-        // Privates
-        private bool drawProcessedMesh = true;
-        private bool isImportActive = false;
         private DateTime timeLastImportedMesh = DateTime.Now;
-        private Mesh customMesh;
+
+        /// <summary>
+        /// For a cached SpatialUnderstanding.Instance.
+        /// </summary>
+        private SpatialUnderstanding spatialUnderstanding;
+
+        /// <summary>
+        /// The spatial understanding mesh will be split into pieces so that we don't have to 
+        /// render the whole mesh, rather just the parts that are visible to the user.
+        /// </summary>
+        private Dictionary<Vector3, MeshData> meshSectors = new Dictionary<Vector3, MeshData>();
+
+        /// <summary>
+        /// A data structure to manage collecting triangles as we 
+        /// subdivide the spatial understanding mesh into smaller sub meshes.
+        /// </summary>
+        class MeshData
```

Comment:
Nit: If this class is public, it should belong in its own file.  If it's private, then make it explicit.